tests/linearizability/linearizability_test.go

```diff
@@ -449,31 +494,28 @@ func persistOperationHistory(t *testing.T, lg *zap.Logger, path string, operatio
 	}
 }
 
-func persistMemberDataDir(t *testing.T, lg *zap.Logger, clus *e2e.EtcdProcessCluster, path string) {
-	for _, member := range clus.Procs {
-		memberDataDir := filepath.Join(path, member.Config().Name)
-		err := os.RemoveAll(memberDataDir)
-		if err != nil {
-			t.Error(err)
-		}
-		lg.Info("Saving member data dir", zap.String("member", member.Config().Name), zap.String("path", memberDataDir))
-		err = os.Rename(member.Config().DataDirPath, memberDataDir)
-		if err != nil {
-			t.Error(err)
-		}
+func persistMemberDataDir(t *testing.T, lg *zap.Logger, member e2e.EtcdProcess, path string) {
+	lg.Info("Saving member data dir", zap.String("member", member.Config().Name), zap.String("path", path))
+	err := os.Rename(member.Config().DataDirPath, path)
+	if err != nil {
+		t.Fatal(err)
 	}
 }
 
-func testResultsDirectory(t *testing.T) (string, error) {
+func testResultsDirectory(t *testing.T) string {
 	path, err := filepath.Abs(filepath.Join(resultsDirectory, strings.ReplaceAll(t.Name(), "/", "_")))
```

Comment:
FYI: 
There was discussion of technique to preserve the t.TempDir file in case of 'catastrophic' test failure:
https://github.com/golang/go/issues/35998#issuecomment-562413587

But I assume you want to preserve the results in case of successful test results as well. 
On flip side - implicit doing it, might lead to getting out of disk space.

Reply:
True, however interrupting tests (ctrl+c) will give the same effect. Difference here that those files are shared between multiple runs so they should not grow in size over time.